webdevcody/code-racer
# Create Profile Page #25

---
Create Profile Page #25
---



## What type of PR is this? (select all that apply)

- [yes ] 🍕 Feature

## Description


I included an initial profile page. It's a first draft of mine, and I welcome anyone to message me who has questions about what I made. I added a function throwError() so we don't have to console.error every error and this func throws an error. Added and exceptions folder under the lib folder as well. Lastly, I added a CSS variable called --monochrome to alternate between black and white depending on light or dark theme, and edited that on tailwindCSS.

## Related Tickets & Documents

- Related Issue # 25
- Closes #

## QA Instructions, Screenshots, Recordings



- Tested it on Windows 11.
- Go to /profile. Since there is no link to that yet, manually input the pathname in the URL.
- Try changing your username! Wait for a few seconds as it is very slow to process right now and we need to add a loading indicator. Only the functionality and initial UI was added.
- Click anywhere outside the modal once you see your name changed (will add a close button soon).
- To view your profile picture, click on it! I use the search Params (check the URL bar) to get the URL of the image.
- The design is responsive. Please message if it's not on your devices.

### UI accessibility concerns?


- Added a time tag just to simplify our lives when adding dates. Probably check if it adheres to accessibility guidelines as I do not know most of the tag's attributes.

## Added/updated tests?

- [check ] 🙋 no, because I need help

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?
